crates/aptos-rest-client/src/lib.rs

```diff
@@ -120,6 +120,27 @@ impl Client {
         })
     }
 
+    pub async fn get_account_balance_at_version(
+        &self,
+        address: AccountAddress,
+        version: u64,
+    ) -> Result<Response<Balance>> {
+        let resp = self
+            .get_account_resource_at_version(
+                address,
+                "0x1::coin::CoinStore<0x1::aptos_coin::AptosCoin>",
+                version,
+            )
+            .await?;
+        resp.and_then(|resource| {
```

Comment:
I'm a little lost here, `and_then` seems to return a result, not an option. I must be missing something obvs though.